protocols/autonatv2/src/client/handler/dial_back.rs

```diff
@@ -1,12 +1,92 @@
-use libp2p_swarm::ConnectionHandler;
+use std::{
+    io,
+    task::{Context, Poll},
+};
 
-#[derive(Debug)]
-pub enum ToBehaviour {
-    GotNonce(u64),
+use either::Either;
+use futures::{AsyncRead, AsyncWrite, AsyncWriteExt};
+use futures_bounded::{FuturesSet, Timeout};
+use libp2p_core::upgrade::{DeniedUpgrade, ReadyUpgrade};
+use libp2p_swarm::{
+    handler::{ConnectionEvent, FullyNegotiatedInbound, ListenUpgradeError},
+    ConnectionHandler, ConnectionHandlerEvent, StreamProtocol, SubstreamProtocol,
+};
+
+use crate::request_response::DialBack;
+
+use super::DEFAULT_TIMEOUT;
+
+pub type ToBehaviour = Result<u64, Either<io::Error, Timeout>>;
```

Comment:
There is `io::ErrorKind::TimedOut` if you wanted to unify them into one error type.

Reply:
I did that